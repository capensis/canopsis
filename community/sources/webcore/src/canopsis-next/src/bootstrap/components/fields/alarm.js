export { default as CAlarmInfosAttributeField } from '@/components/forms/fields/alarm/c-alarm-infos-attribute-field.vue';
export { default as CAlarmTagField } from '@/components/forms/fields/alarm/c-alarm-tag-field.vue';
export { default as CAlarmField } from '@/components/forms/fields/alarm/c-alarm-field.vue';
