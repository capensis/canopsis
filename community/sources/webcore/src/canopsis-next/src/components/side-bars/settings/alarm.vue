<template lang="pug">
  div
    v-list.pt-0(expand)
      field-title(v-model="settings.widget.title", :title="$t('common.title')")
      v-divider
      field-periodic-refresh(v-model="settings.widget.parameters.periodic_refresh")
      v-divider
      v-list-group(data-test="advancedSettings")
        v-list-tile(slot="activator") {{ $t('settings.advancedSettings') }}
        v-list.grey.lighten-4.px-2.py-0(expand)
          field-default-sort-column(
            v-model="settings.widget.parameters.sort",
            :columns="settings.widget.parameters.widgetColumns",
            :columns-label="$t('settings.columnName')"
          )
          v-divider
          field-columns(
            v-model="settings.widget.parameters.widgetColumns",
            :label="$t('settings.columnNames')",
            with-html,
            with-color-indicator
          )
          v-divider
          field-columns(
            v-model="settings.widget.parameters.widgetGroupColumns",
            :label="$t('settings.groupColumnNames')",
            with-html,
            with-color-indicator
          )
          v-divider
          field-columns(
            v-model="settings.widget.parameters.serviceDependenciesColumns",
            :label="$t('settings.trackColumnNames')",
            with-color-indicator
          )
          v-divider
          field-default-elements-per-page(v-model="settings.userPreferenceContent.itemsPerPage")
          v-divider
          field-opened-resolved-filter(v-model="settings.widget.parameters.opened")
          v-divider
          template(v-if="hasAccessToListFilters")
            field-filters(
              v-model="settings.widget.parameters.mainFilter",
              :entities-type="$constants.ENTITIES_TYPES.alarm",
              :filters.sync="settings.widget.parameters.viewFilters",
              :condition.sync="settings.widget.parameters.mainFilterCondition",
              :has-access-to-add-filter="hasAccessToAddFilter",
              :has-access-to-edit-filter="hasAccessToEditFilter",
              @input="updateMainFilterUpdatedAt"
            )
            v-divider
          template(v-if="hasAccessToListRemediationInstructionsFilters")
            field-remediation-instructions-filters(
              v-model="settings.widget.parameters.remediationInstructionsFilters",
              :has-access-to-add-filter="hasAccessToAddRemediationInstructionsFilter",
              :has-access-to-edit-filter="hasAccessToEditRemediationInstructionsFilter"
            )
            v-divider
          field-live-reporting(v-model="settings.widget.parameters.liveReporting")
          v-divider
          field-info-popup(
            v-model="settings.widget.parameters.infoPopups",
            :columns="settings.widget.parameters.widgetColumns"
          )
          v-divider
          field-text-editor(
            data-test="widgetMoreInfoTemplate",
            v-model="settings.widget.parameters.moreInfoTemplate",
            :title="$t('settings.moreInfosModal')"
          )
          v-divider
          field-grid-range-size(
            v-model="settings.widget.parameters.expandGridRangeSize",
            :title="$t('settings.expandGridRangeSize')"
          )
          v-divider
          field-switcher(
            data-test="isHtmlEnabledOnTimeLine",
            v-model="settings.widget.parameters.isHtmlEnabledOnTimeLine",
            :title="$t('settings.isHtmlEnabledOnTimeLine')"
          )
          v-divider
          v-list-group(data-test="ackGroup")
            v-list-tile(slot="activator") Ack
            v-list.grey.lighten-4.px-2.py-0(expand)
            field-switcher(
              data-test="isAckNoteRequired",
              v-model="settings.widget.parameters.isAckNoteRequired",
              :title="$t('settings.isAckNoteRequired')"
            )
            v-divider
            field-switcher(
              data-test="isMultiAckEnabled",
              v-model="settings.widget.parameters.isMultiAckEnabled",
              :title="$t('settings.isMultiAckEnabled')"
            )
            v-divider
            field-fast-ack-output(v-model="settings.widget.parameters.fastAckOutput")
          v-divider
          field-switcher(
            v-model="settings.widget.parameters.isSnoozeNoteRequired",
            :title="$t('settings.isSnoozeNoteRequired')"
          )
          v-divider
          field-enabled-limit(
            v-model="settings.widget.parameters.linksCategoriesAsList",
            :title="$t('settings.linksCategoriesAsList')",
            :label="$t('settings.linksCategoriesLimit')"
          )
          v-divider
          export-csv-form(v-model="settings.widget.parameters", datetime-format)
          v-divider
          field-switcher(
            v-model="settings.widget.parameters.sticky_header",
            :title="$t('settings.stickyHeader')"
          )
      v-divider
    v-btn.primary(data-test="submitAlarms", @click="submit") {{ $t('common.save') }}
</template>

<script>
import { get } from 'lodash';

import { PAGINATION_LIMIT } from '@/config';
import { SIDE_BARS } from '@/constants';

import { alarmListWidgetToForm, formToAlarmListWidget } from '@/helpers/forms/widgets/alarm';

import { widgetSettingsMixin } from '@/mixins/widget/settings';
import { permissionsWidgetsAlarmsListFilters } from '@/mixins/permissions/widgets/alarms-list/filters';
import { permissionsWidgetsAlarmsListRemediationInstructionsFilters }
  from '@/mixins/permissions/widgets/alarms-list/remediation-instructions-filters';

import FieldTitle from './fields/common/title.vue';
import FieldDefaultSortColumn from './fields/common/default-sort-column.vue';
import FieldColumns from './fields/common/columns.vue';
import FieldLiveReporting from './fields/common/live-reporting.vue';
import FieldPeriodicRefresh from './fields/common/periodic-refresh.vue';
import FieldDefaultElementsPerPage from './fields/common/default-elements-per-page.vue';
import FieldFilters from './fields/common/filters.vue';
import FieldTextEditor from './fields/common/text-editor.vue';
import FieldSwitcher from './fields/common/switcher.vue';
import FieldFastAckOutput from './fields/alarm/fast-ack-output.vue';
import FieldGridRangeSize from './fields/common/grid-range-size.vue';
import FieldRemediationInstructionsFilters from './fields/common/remediation-instructions-filters.vue';
import FieldOpenedResolvedFilter from './fields/alarm/opened-resolved-filter.vue';
import FieldInfoPopup from './fields/alarm/info-popup.vue';
import FieldEnabledLimit from './fields/common/enabled-limit.vue';
import ExportCsvForm from './forms/export-csv.vue';

/**
 * Component to regroup the alarms list settings fields
 */
export default {
  name: SIDE_BARS.alarmSettings,
  $_veeValidate: {
    validator: 'new',
  },
  components: {
    FieldTitle,
    FieldDefaultSortColumn,
    FieldColumns,
    FieldLiveReporting,
    FieldPeriodicRefresh,
    FieldDefaultElementsPerPage,
    FieldOpenedResolvedFilter,
    FieldFilters,
    FieldTextEditor,
    FieldSwitcher,
    FieldFastAckOutput,
    FieldGridRangeSize,
    FieldRemediationInstructionsFilters,
    FieldInfoPopup,
    FieldEnabledLimit,
    ExportCsvForm,
  },
  mixins: [
    widgetSettingsMixin,
    permissionsWidgetsAlarmsListFilters,
    permissionsWidgetsAlarmsListRemediationInstructionsFilters,
  ],
  data() {
    const { widget } = this.config;

    return {
      settings: {
        widget: alarmListWidgetToForm(widget),
        userPreferenceContent: { itemsPerPage: PAGINATION_LIMIT },
      },
    };
  },
  mounted() {
    const { content } = this.userPreference;

    this.settings.userPreferenceContent = {
      itemsPerPage: get(content, 'itemsPerPage', PAGINATION_LIMIT),
    };
  },
  methods: {
    updateMainFilterUpdatedAt() {
      this.settings.widget.parameters.mainFilterUpdatedAt = Date.now();
    },

    prepareWidgetSettings() {
      const { widget } = this.settings;

      return formToAlarmListWidget(widget);
    },
  },
};
</script>
