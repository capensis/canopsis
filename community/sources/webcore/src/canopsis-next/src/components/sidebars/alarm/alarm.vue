<template lang="pug">
  widget-settings(:submitting="submitting", @submit="submit")
    field-title(v-model="form.title")
    v-divider
    field-periodic-refresh(v-model="form.parameters.periodic_refresh")
    v-divider
    widget-settings-group(:title="$t('settings.advancedSettings')")
      field-default-sort-column(
        v-model="form.parameters.sort",
        :columns="sortablePreparedWidgetColumns",
        :columns-label="$t('settings.columnName')"
      )
      v-divider
      field-columns(
        v-model="form.parameters.widgetColumns",
        :template="form.parameters.widgetColumnsTemplate",
        :templates="alarmColumnsWidgetTemplates",
        :templates-pending="widgetTemplatesPending",
        :label="$t('settings.columnNames')",
        :type="$constants.ENTITIES_TYPES.alarm",
        with-template,
        with-html,
        with-color-indicator,
        @update:template="updateWidgetColumnsTemplate"
      )
      v-divider
      field-resize-column-behavior(v-model="form.parameters.columns")
      v-divider
      field-columns(
        v-model="form.parameters.widgetGroupColumns",
        :template="form.parameters.widgetGroupColumnsTemplate",
        :templates="alarmColumnsWidgetTemplates",
        :templates-pending="widgetTemplatesPending",
        :label="$t('settings.groupColumnNames')",
        :type="$constants.ENTITIES_TYPES.alarm",
        with-html,
        with-color-indicator,
        @update:template="updateWidgetGroupColumnsTemplate"
      )
      v-divider
      field-columns(
        v-model="form.parameters.serviceDependenciesColumns",
        :template="form.parameters.serviceDependenciesColumnsTemplate",
        :templates="entityColumnsWidgetTemplates",
        :templates-pending="widgetTemplatesPending",
        :label="$t('settings.trackColumnNames')",
        :type="$constants.ENTITIES_TYPES.entity",
        with-color-indicator,
        @update:template="updateServiceDependenciesColumnsTemplate"
      )
      v-divider
      field-default-elements-per-page(v-model="form.parameters.itemsPerPage")
      v-divider
      field-density(v-model="form.parameters.dense")
      v-divider
      field-opened-resolved-filter(v-model="form.parameters.opened")
      v-divider
      template(v-if="hasAccessToListFilters")
        field-filters(
          v-model="form.parameters.mainFilter",
          :filters.sync="form.filters",
          :widget-id="widget._id",
          :addable="hasAccessToAddFilter",
          :editable="hasAccessToEditFilter",
          with-alarm,
          with-entity,
          with-pbehavior
        )
        v-divider
      template(v-if="hasAccessToListRemediationInstructionsFilters")
        field-remediation-instructions-filters(
          v-model="form.parameters.remediationInstructionsFilters",
          :addable="hasAccessToAddRemediationInstructionsFilter",
          :editable="hasAccessToEditRemediationInstructionsFilter"
        )
        v-divider
      field-switcher(
        v-model="form.parameters.clearFilterDisabled",
        :title="$t('settings.clearFilterDisabled')"
      )
      v-divider
      field-live-reporting(v-model="form.parameters.liveReporting")
      v-divider
      field-info-popup(
        v-model="form.parameters.infoPopups",
        :columns="preparedWidgetColumns"
      )
      v-divider
      field-text-editor-with-template(
        :value="form.parameters.moreInfoTemplate",
        :template="form.parameters.moreInfoTemplateTemplate",
        :title="$t('settings.moreInfosModal')",
        :variables="alarmVariables",
        :templates="alarmMoreInfosWidgetTemplates",
        addable,
        removable,
        @input="updateMoreInfo"
      )
      v-divider
      field-grid-range-size(
        v-model="form.parameters.expandGridRangeSize",
        :title="$t('settings.expandGridRangeSize')"
      )
      v-divider
      field-switcher(
        v-model="form.parameters.isHtmlEnabledOnTimeLine",
        :title="$t('settings.isHtmlEnabledOnTimeLine')"
      )
      v-divider
      widget-settings-group(:title="$t('common.ack')")
        field-switcher(
          v-model="form.parameters.isAckNoteRequired",
          :title="$t('settings.isAckNoteRequired')"
        )
        v-divider
        field-switcher(
          v-model="form.parameters.isMultiAckEnabled",
          :title="$t('settings.isMultiAckEnabled')"
        )
        v-divider
        field-fast-action-output(
          v-model="form.parameters.fastAckOutput",
          :label="$t('settings.fastAckOutput')"
        )
      v-divider
      widget-settings-group(:title="$t('common.cancel')")
        field-fast-action-output(
          v-model="form.parameters.fastCancelOutput",
          :label="$t('settings.fastCancelOutput')"
        )
      v-divider
      field-switcher(
        v-model="form.parameters.isSnoozeNoteRequired",
        :title="$t('settings.isSnoozeNoteRequired')"
      )
      v-divider
      field-switcher(
        v-model="form.parameters.isMultiDeclareTicketEnabled",
        :title="$t('settings.isMultiDeclareTicketEnabled')"
      )
      v-divider
      field-number(
        v-model="form.parameters.inlineLinksCount",
        :title="$t('settings.inlineLinksCount')"
      )
      v-divider
      export-csv-form(
        v-model="form.parameters",
        :type="$constants.ENTITIES_TYPES.alarm",
        :templates="alarmColumnsWidgetTemplates",
        :templates-pending="widgetTemplatesPending",
        datetime-format
      )
      v-divider
      field-switcher(
        v-model="form.parameters.sticky_header",
        :title="$t('settings.stickyHeader')"
      )
      v-divider
      widget-settings-group(:title="$t('settings.kioskMode')")
        field-switcher(
          v-model="form.parameters.kiosk.hideActions",
          :title="$t('settings.kiosk.hideActions')"
        )
        v-divider
        field-switcher(
          v-model="form.parameters.kiosk.hideMassSelection",
          :title="$t('settings.kiosk.hideMassSelection')"
        )
        v-divider
        field-switcher(
          v-model="form.parameters.kiosk.hideToolbar",
          :title="$t('settings.kiosk.hideToolbar')"
        )
    v-divider
    charts-form(v-model="form.parameters.charts")
    v-divider
</template>

<script>
import { SIDE_BARS, ALARM_UNSORTABLE_FIELDS, ALARM_FIELDS_TO_LABELS_KEYS } from '@/constants';

import { formToWidgetColumns } from '@/helpers/entities/widget/column/form';
import { getWidgetColumnLabel, getWidgetColumnSortable } from '@/helpers/entities/widget/list';

import { widgetSettingsMixin } from '@/mixins/widget/settings';
import { entitiesInfosMixin } from '@/mixins/entities/infos';
import { alarmVariablesMixin } from '@/mixins/widget/variables';
import { widgetTemplatesMixin } from '@/mixins/widget/templates';
import { permissionsWidgetsAlarmsListFilters } from '@/mixins/permissions/widgets/alarms-list/filters';
import { permissionsWidgetsAlarmsListRemediationInstructionsFilters }
  from '@/mixins/permissions/widgets/alarms-list/remediation-instructions-filters';

import WidgetSettingsGroup from '../partials/widget-settings-group.vue';
import WidgetSettings from '../partials/widget-settings.vue';
import FieldTitle from '../form/fields/title.vue';
import FieldDefaultSortColumn from '../form/fields/default-sort-column.vue';
import FieldColumns from '../form/fields/columns.vue';
import FieldPeriodicRefresh from '../form/fields/periodic-refresh.vue';
import FieldDefaultElementsPerPage from '../form/fields/default-elements-per-page.vue';
import FieldFilters from '../form/fields/filters.vue';
import FieldTextEditorWithTemplate from '../form/fields/text-editor-with-template.vue';
import FieldSwitcher from '../form/fields/switcher.vue';
import FieldNumber from '../form/fields/number.vue';
import ExportCsvForm from '../form/export-csv.vue';
import ChartsForm from '../chart/form/charts-form.vue';

import FieldGridRangeSize from './form/fields/grid-range-size.vue';
import FieldRemediationInstructionsFilters from './form/fields/remediation-instructions-filters.vue';
import FieldDensity from './form/fields/density.vue';
import FieldLiveReporting from './form/fields/live-reporting.vue';
import FieldFastActionOutput from './form/fields/fast-action-output.vue';
import FieldOpenedResolvedFilter from './form/fields/opened-resolved-filter.vue';
import FieldInfoPopup from './form/fields/info-popup.vue';
import FieldResizeColumnBehavior from './form/fields/resize-column-behavior.vue';

/**
 * Component to regroup the alarms list settings fields
 */
export default {
  name: SIDE_BARS.alarmSettings,
  components: {
    WidgetSettingsGroup,
    WidgetSettings,
    FieldTitle,
    FieldDefaultSortColumn,
    FieldColumns,
    FieldLiveReporting,
    FieldPeriodicRefresh,
    FieldDefaultElementsPerPage,
    FieldOpenedResolvedFilter,
    FieldFilters,
    FieldTextEditorWithTemplate,
    FieldSwitcher,
    FieldFastActionOutput,
    FieldGridRangeSize,
    FieldRemediationInstructionsFilters,
    FieldInfoPopup,
    FieldDensity,
    FieldNumber,
    ExportCsvForm,
    ChartsForm,
    FieldResizeColumnBehavior,
  },
  mixins: [
    widgetSettingsMixin,
    entitiesInfosMixin,
    alarmVariablesMixin,
    widgetTemplatesMixin,
    permissionsWidgetsAlarmsListFilters,
    permissionsWidgetsAlarmsListRemediationInstructionsFilters,
  ],
  computed: {
    preparedWidgetColumns() {
      return formToWidgetColumns(this.form.parameters.widgetColumns).map(column => ({
        ...column,

        text: getWidgetColumnLabel(column, ALARM_FIELDS_TO_LABELS_KEYS),
      }));
    },

    sortablePreparedWidgetColumns() {
      return this.preparedWidgetColumns.filter(column => getWidgetColumnSortable(column, ALARM_UNSORTABLE_FIELDS));
    },
  },
  mounted() {
    this.fetchInfos();
  },
  methods: {
    updateWidgetColumnsTemplate(template, columns) {
      this.$set(this.form.parameters, 'widgetColumnsTemplate', template);
      this.$set(this.form.parameters, 'widgetColumns', columns);
    },

    updateWidgetGroupColumnsTemplate(template, columns) {
      this.$set(this.form.parameters, 'widgetGroupColumnsTemplate', template);
      this.$set(this.form.parameters, 'widgetGroupColumns', columns);
    },

    updateServiceDependenciesColumnsTemplate(template, columns) {
      this.$set(this.form.parameters, 'serviceDependenciesColumnsTemplate', template);
      this.$set(this.form.parameters, 'serviceDependenciesColumns', columns);
    },

    updateMoreInfo(content, template) {
      this.$set(this.form.parameters, 'moreInfoTemplate', content);

      if (template && template !== this.form.parameters.moreInfoTemplateTemplate) {
        this.$set(this.form.parameters, 'moreInfoTemplateTemplate', template);
      }
    },
  },
};
</script>
