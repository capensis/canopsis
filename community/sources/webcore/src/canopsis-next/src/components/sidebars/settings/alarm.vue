<template lang="pug">
  div
    v-list.pt-0(expand)
      field-title(v-model="form.title", :title="$t('common.title')")
      v-divider
      field-periodic-refresh(v-model="form.parameters.periodic_refresh")
      v-divider
      v-list-group
        template(#activator="")
          v-list-tile {{ $t('settings.advancedSettings') }}
        v-list.grey.lighten-4.px-2.py-0(expand)
          field-default-sort-column(
            v-model="form.parameters.sort",
            :columns="form.parameters.widgetColumns",
            :columns-label="$t('settings.columnName')"
          )
          v-divider
          field-columns(
            v-model="form.parameters.widgetColumns",
            :label="$t('settings.columnNames')",
            with-html,
            with-template,
            with-color-indicator
          )
          v-divider
          field-columns(
            v-model="form.parameters.widgetGroupColumns",
            :label="$t('settings.groupColumnNames')",
            with-html,
            with-color-indicator
          )
          v-divider
          field-columns(
            v-model="form.parameters.serviceDependenciesColumns",
            :label="$t('settings.trackColumnNames')",
            with-color-indicator
          )
          v-divider
          field-default-elements-per-page(v-model="form.parameters.itemsPerPage")
          v-divider
          field-density(v-model="form.parameters.dense")
          v-divider
          field-opened-resolved-filter(v-model="form.parameters.opened")
          v-divider
          template(v-if="hasAccessToListFilters")
            field-filters(
              v-model="form.parameters.mainFilter",
              :entities-type="$constants.ENTITIES_TYPES.alarm",
              :filters.sync="form.parameters.viewFilters",
              :condition.sync="form.parameters.mainFilterCondition",
              :addable="hasAccessToAddFilter",
              :editable="hasAccessToEditFilter",
              @input="updateMainFilterUpdatedAt"
            )
            v-divider
          template(v-if="hasAccessToListRemediationInstructionsFilters")
            field-remediation-instructions-filters(
              v-model="form.parameters.remediationInstructionsFilters",
              :addable="hasAccessToAddRemediationInstructionsFilter",
              :editable="hasAccessToEditRemediationInstructionsFilter"
            )
            v-divider
          field-live-reporting(v-model="form.parameters.liveReporting")
          v-divider
          field-info-popup(
            v-model="form.parameters.infoPopups",
            :columns="form.parameters.widgetColumns"
          )
          v-divider
          field-text-editor(
            v-model="form.parameters.moreInfoTemplate",
            :title="$t('settings.moreInfosModal')"
          )
          v-divider
          field-grid-range-size(
            v-model="form.parameters.expandGridRangeSize",
            :title="$t('settings.expandGridRangeSize')"
          )
          v-divider
          field-switcher(
            v-model="form.parameters.isHtmlEnabledOnTimeLine",
            :title="$t('settings.isHtmlEnabledOnTimeLine')"
          )
          v-divider
          v-list-group
            template(#activator="")
              v-list-tile Ack
            v-list.grey.lighten-4.px-2.py-0(expand)
              field-switcher(
                v-model="form.parameters.isAckNoteRequired",
                :title="$t('settings.isAckNoteRequired')"
              )
              v-divider
              field-switcher(
                v-model="form.parameters.isMultiAckEnabled",
                :title="$t('settings.isMultiAckEnabled')"
              )
              v-divider
              field-fast-ack-output(v-model="form.parameters.fastAckOutput")
          v-divider
          field-switcher(
            v-model="form.parameters.isSnoozeNoteRequired",
            :title="$t('settings.isSnoozeNoteRequired')"
          )
          v-divider
          field-enabled-limit(
            v-model="form.parameters.linksCategoriesAsList",
            :title="$t('settings.linksCategoriesAsList')",
            :label="$t('settings.linksCategoriesLimit')"
          )
          v-divider
          export-csv-form(v-model="form.parameters", datetime-format)
          v-divider
          field-switcher(
            v-model="form.parameters.sticky_header",
            :title="$t('settings.stickyHeader')"
          )
      v-divider
    v-btn.primary(@click="submit") {{ $t('common.save') }}
</template>

<script>
import { SIDE_BARS } from '@/constants';

import { widgetSettingsMixin } from '@/mixins/widget/settings';
import { permissionsWidgetsAlarmsListFilters } from '@/mixins/permissions/widgets/alarms-list/filters';
import { permissionsWidgetsAlarmsListRemediationInstructionsFilters }
  from '@/mixins/permissions/widgets/alarms-list/remediation-instructions-filters';

import FieldTitle from './fields/common/title.vue';
import FieldDefaultSortColumn from './fields/common/default-sort-column.vue';
import FieldColumns from './fields/common/columns.vue';
import FieldLiveReporting from './fields/common/live-reporting.vue';
import FieldPeriodicRefresh from './fields/common/periodic-refresh.vue';
import FieldDefaultElementsPerPage from './fields/common/default-elements-per-page.vue';
import FieldFilters from './fields/common/filters.vue';
import FieldTextEditor from './fields/common/text-editor.vue';
import FieldSwitcher from './fields/common/switcher.vue';
import FieldFastAckOutput from './fields/alarm/fast-ack-output.vue';
import FieldGridRangeSize from './fields/common/grid-range-size.vue';
import FieldRemediationInstructionsFilters from './fields/common/remediation-instructions-filters.vue';
import FieldOpenedResolvedFilter from './fields/alarm/opened-resolved-filter.vue';
import FieldInfoPopup from './fields/alarm/info-popup.vue';
import FieldEnabledLimit from './fields/common/enabled-limit.vue';
import FieldDensity from './fields/common/density.vue';
import ExportCsvForm from './forms/export-csv.vue';

/**
 * Component to regroup the alarms list settings fields
 */
export default {
  name: SIDE_BARS.alarmSettings,
  components: {
    FieldTitle,
    FieldDefaultSortColumn,
    FieldColumns,
    FieldLiveReporting,
    FieldPeriodicRefresh,
    FieldDefaultElementsPerPage,
    FieldOpenedResolvedFilter,
    FieldFilters,
    FieldTextEditor,
    FieldSwitcher,
    FieldFastAckOutput,
    FieldGridRangeSize,
    FieldRemediationInstructionsFilters,
    FieldInfoPopup,
    FieldEnabledLimit,
    FieldDensity,
    ExportCsvForm,
  },
  mixins: [
    widgetSettingsMixin,
    permissionsWidgetsAlarmsListFilters,
    permissionsWidgetsAlarmsListRemediationInstructionsFilters,
  ],
};
</script>
