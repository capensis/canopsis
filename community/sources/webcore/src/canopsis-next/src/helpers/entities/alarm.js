import { ENTITIES_STATES, ENTITIES_STATUSES } from '@/constants';

/**
 * Check alarm state is ok
 *
 * @param {Alarm} alarm
 * @returns {boolean}
 */
export const isAlarmStateOk = alarm => alarm.v?.state?.val === ENTITIES_STATES.ok;

/**
 * Check alarm status is closed
 *
 * @param {Alarm} alarm
 * @returns {boolean}
 */
export const isAlarmStatusClosed = alarm => alarm.v?.status?.val === ENTITIES_STATUSES.closed;

/**
 * Check alarm status is cancelled
 *
 * @param {Alarm} alarm
 * @returns {boolean}
 */
export const isAlarmStatusCancelled = alarm => alarm.v?.status?.val === ENTITIES_STATUSES.cancelled;

/**
 * Check alarm status is ongoing
 *
 * @param {Alarm} alarm
 * @returns {boolean}
 */
export const isAlarmStatusOngoing = alarm => alarm.v?.status?.val === ENTITIES_STATUSES.ongoing;

/**
 * Check alarm status is flapping
 *
 * @param {Alarm} alarm
 * @returns {boolean}
 */
export const isAlarmStatusFlapping = alarm => alarm.v?.status?.val === ENTITIES_STATUSES.flapping;
