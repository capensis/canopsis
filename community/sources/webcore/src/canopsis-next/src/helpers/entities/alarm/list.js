import { groupBy } from 'lodash';

import {
  ALARM_LINK_ICON_CHIP_COLUMN_GAP,
  ALARM_LINK_ICON_CHIP_WIDTH,
  ALARM_LINK_TD_PADDINGS,
  DATETIME_FORMATS,
} from '@/constants';

import { convertDateToString } from '@/helpers/date/date';

/**
 * @typedef {0 | 1 | 2} AlarmListWidgetDense
 */

/**
 * Generate alarm details id by widgetId
 *
 * @param {string} alarmId
 * @param {string} widgetId
 * @returns {string}
 */
export const generateAlarmDetailsId = (alarmId, widgetId) => `${alarmId}_${widgetId}`;

/**
 * Get dataPreparer for alarmDetails entity
 *
 * @param {string} widgetId
 * @returns {Function}
 */
export const getAlarmDetailsDataPreparer = widgetId => data => (
  data.map(item => ({
    ...item,

    /**
     * We are generating new id based on alarmId and widgetId to avoiding collision with two widgets
     * on the same view with opened expand panel on the same alarm
     */
    _id: generateAlarmDetailsId(item._id, widgetId),
  }))
);

/**
 * Get grouped steps by date
 *
 * @param {AlarmEvent[]} steps
 * @return {Object.<string, AlarmEvent[]>}
 */
export const groupAlarmSteps = steps => (
  groupBy(steps, step => convertDateToString(step.t, DATETIME_FORMATS.short))
);

/**
 * Calculate alarm links column width for table
 *
 * @param {AlarmListWidgetDense} dense
 * @param {number} linksInRowCount
 * @returns {`${number}px`}
 */
export const calculateAlarmLinksColumnWidth = (dense, linksInRowCount) => (
  `${(ALARM_LINK_ICON_CHIP_WIDTH * linksInRowCount)
  + (ALARM_LINK_ICON_CHIP_COLUMN_GAP[dense] * (linksInRowCount - 1))
  + (ALARM_LINK_TD_PADDINGS[dense] * 2)}px`
);
