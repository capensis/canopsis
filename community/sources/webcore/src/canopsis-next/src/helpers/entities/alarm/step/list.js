import { groupBy } from 'lodash';

import { DATETIME_FORMATS } from '@/constants';

import { convertDateToString } from '@/helpers/date/date';
import { addKeyInEntities } from '@/helpers/array';

/**
 * Get grouped steps by date
 *
 * @param {AlarmStep[]} steps
 * @return {Object.<string, AlarmStep[]>}
 */
export const groupAlarmSteps = steps => (
  groupBy(addKeyInEntities(steps), step => convertDateToString(step.t, DATETIME_FORMATS.short))
);
