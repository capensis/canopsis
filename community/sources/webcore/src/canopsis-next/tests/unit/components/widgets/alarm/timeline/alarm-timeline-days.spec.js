import { generateRenderer } from '@unit/utils/vue';
import { fakeAlarmDetails } from '@unit/data/alarm';

import { groupAlarmSteps } from '@/helpers/entities/alarm/step/list';

import AlarmTimelineDays from '@/components/widgets/alarm/timeline/alarm-timeline-days.vue';

const stubs = {
  'alarm-timeline-steps': true,
};

describe('alarm-timeline-days', () => {
  const { steps: { data: steps } } = fakeAlarmDetails();
  const snapshotFactory = generateRenderer(AlarmTimelineDays, { stubs });

  test('Renders `alarm-timeline-days` correctly', () => {
    const wrapper = snapshotFactory({
      propsData: {
        days: groupAlarmSteps(steps),
      },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
