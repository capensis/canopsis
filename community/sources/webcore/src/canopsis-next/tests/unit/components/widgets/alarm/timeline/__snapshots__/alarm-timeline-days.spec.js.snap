// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alarm-timeline-days Renders \`alarm-timeline-days\` correctly 1`] = `
<div
  class="layout column"
>
  <div
    class="layout my-1 column"
  >
    <span
      class="grey--text my-2"
    >
      13/07/2021
    </span>
    <alarm-timeline-steps-stub
      steps="[object Object],[object Object]"
    />
  </div>
  <div
    class="layout my-1 column"
  >
    <span
      class="grey--text my-2"
    >
      30/07/2021
    </span>
    <alarm-timeline-steps-stub
      steps="[object Object]"
    />
  </div>
  <div
    class="layout my-1 column"
  >
    <span
      class="grey--text my-2"
    >
      27/09/2021
    </span>
    <alarm-timeline-steps-stub
      steps="[object Object],[object Object]"
    />
  </div>
  <div
    class="layout my-1 column"
  >
    <span
      class="grey--text my-2"
    >
      30/09/2021
    </span>
    <alarm-timeline-steps-stub
      steps="[object Object]"
    />
  </div>
  <div
    class="layout my-1 column"
  >
    <span
      class="grey--text my-2"
    >
      18/10/2021
    </span>
    <alarm-timeline-steps-stub
      steps="[object Object]"
    />
  </div>
</div>
`;
